modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalDeleteObjectDefinition.tsx

```diff
@@ -0,0 +1,197 @@
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {ClayModalProvider, useModal} from '@clayui/modal';
+import {Observer} from '@clayui/modal/lib/types';
+import {createResourceURL, fetch} from 'frontend-js-web';
+import React, {useEffect, useState} from 'react';
+
+import DangerModal from './DangerModal';
+import WarningModal from './WarningModal';
+
+function ModalDeleteObjectDefinition({
+	objectDefinition,
+	observer,
+	onClose,
+	onDelete,
+}: IProps) {
+	const {
+		hasObjectRelationship,
+		id,
+		name,
+		objectEntriesCount,
+		status: {code},
+	} = objectDefinition;
+
+	return hasObjectRelationship ? (
+		<WarningModal
+			observer={observer}
+			onClose={onClose}
+			title={Liferay.Language.get('deletion-not-allowed')}
+		>
+			<div>
+				{Liferay.Language.get(
+					'this-object-cannot-be-deleted-because-it-is-related-to-others'
+				)}
+			</div>
+
+			<div>
+				{Liferay.Language.get(
+					'to-delete-this-object-you-need-first-delete-its-relationships'
+				)}
+			</div>
+
+			<div>
+				{Liferay.Language.get('go-to-object-details-relationships')}
+			</div>
+		</WarningModal>
+	) : code === 0 ? (
+		<DangerModal
+			errorMessage={Liferay.Language.get(
+				'input-and-object-name-does-not-match'
+			)}
+			observer={observer}
+			onClose={onClose}
+			onDelete={() => onDelete(id)}
+			title={Liferay.Language.get('delete-custom-object')}
+			token={name}
+		>
+			<p>
+				{Liferay.Language.get(
+					'deleted-custom-objects-cannot-be-restored'
+				)}
+			</p>
+
+			<ul>
+				<li>{Liferay.Language.get('objects-definitions')}</li>
+
+				<li>
+					{Liferay.Language.get('all-data-records-in-the-object')}
+				</li>
+			</ul>
+
+			<p>
+				{Liferay.Util.sub(
+					Liferay.Language.get('x-has-x-records'),
+					`${name}`,
+					`${objectEntriesCount}`
+				)}
+			</p>
+
+			<p>
+				{Liferay.Language.get(
+					'to-export-all-records-go-to-global-menu-applications-import-export-center'
+				)}
+			</p>
+
+			<p
+				dangerouslySetInnerHTML={{
+					__html: Liferay.Util.sub(
+						Liferay.Language.get(
+							'please-type-the-object-name-x-to-confirm'
+						),
+						`<strong>${name}</strong>`
+					),
+				}}
+			/>
+		</DangerModal>
+	) : (
+		onDelete(id)
+	);
+}
+
+interface IProps {
+	objectDefinition: any;
+	observer: Observer;
+	onClose: () => void;
+	onDelete: any;
+}
+
+export default function ModalWithProvider({baseResourceURL}: any) {
+	const [objectDefinition, setObjectDefinition] = useState<any>(null);
+
+	const getDeleteObjectDefinition = async ({itemData}: any) => {
```

Comment:
Avoid using the `any` type, be as strict as possible

If you have no idea about the type to use, you better start using `unknown` type instead:
```suggestion
export default function ModalWithProvider({baseResourceURL}: {baseResourceURL: unknown}) {
	const [objectDefinition, setObjectDefinition] = useState(null);

	const getDeleteObjectDefinition = async ({itemData}: {itemData: unknown}) => {
```